workshops/util.py

```diff
@@ -28,3 +66,84 @@ def earth_distance(pos1, pos2):
 
     # Multiply by 6373 to get distance in km.
     return arc * 6373
+
+
+def upload_person_task_csv(uploaded_file):
+    """
+    Read data from CSV and turn it into JSON-serializable list of dictionaries.
+    "Serializability" is required because we put this data into session.  See
+    https://docs.djangoproject.com/en/1.7/topics/http/sessions/ for details.
+
+    Also return a list of fields from Person.PERSON_UPLOAD_FIELDS for which
+    no data was given.
+    """
+    persons_tasks = []
+    reader = csv.DictReader(uploaded_file)
+    empty_fields = []
+    for row in reader:
+        person_fields= {}
+        for col in Person.PERSON_UPLOAD_FIELDS:
+            try:
+                person_fields[col] = row[col].strip()
+            except KeyError:
+                if col not in empty_fields:
+                    empty_fields.append(col)
+        entry = {'person': person_fields, 'event': row.get('event', None),
+                 'role': row.get('role', None), 'errors': None}
+        persons_tasks.append(entry)
+    return persons_tasks, empty_fields
+
+
+def verify_upload_person_task(data):
+    """
+    Verify that uploaded data is correct.  Show errors by populating ``errors``
+    dictionary item.
+
+    This function changes ``data`` in place; ``data`` is a list, so it's
+    passed by a reference.  This means any changes to ``data`` we make in
+    here don't need to be returned via ``return`` statement.
+    """
+
+    errors_occur = False
+
+    for item in data:
+        errors = []
+        event, role = item.get('event', None), item.get('role', None)
+        try:
+            email = item['person'].get('email', None)
+        except KeyError:
+            email = None
+            errors.append("'person' key not in item")
+
+        if event:
+            try:
+                Event.objects.get(slug=event)
+            except Event.DoesNotExist:
+                errors.append('Event with slug {} does not exist.'
+                              .format(event))
+        if role:
+            try:
+                Role.objects.get(name=role)
+            except Role.DoesNotExist:
+                errors.append('Role with name {} does not exist.'.format(role))
+            except Role.MultipleObjectsReturned:
+                errors.append('More than one role named {} exists.'
```

Comment:
Strings in lines 122, 128 and 130 (`errors.append(…)`) should be unicode (ie. `u'Event with slug…'`). Otherwise when uploading non-ASCII characters there would be errors in these lines.